File: usacobronze/2021jan23/prob3.py
n=int(input())
c,s=list(sorted(map(int,input().split()))),list(map(int,sorted(input().split())))

d={x:[y for y in s if x<=y] for x in reversed(c)}
lens=[len(d[x]) for x in d]
prod=1
takeout=0
for x in lens:
    prod*=x-takeout
    takeout+=1
print(prod)

File: usacobronze/2021jan23/prob3.cpp
#include <algorithm>
#include <iostream>

using namespace std;
bool gr(int a, int b) {return a>b;}
int main() {
    int n,len,takeout=0;
    cin >> n;
    int c[n],s[n];
    for (int i=0;i<n;i++) cin >> c[i];
    for (int i=0;i<n;i++) cin >> s[i];
    long long prod=1;
    sort(c,c+n,gr);
    sort(s,s+n);
    for (int i: c) {
        len=0;
        for (int y: s) if (i<=y) len++;
        prod*=len-takeout++;
    }
    cout << prod << endl;
}